addons/base_vat/models/res_partner.py

```diff
@@ -112,30 +114,44 @@ def simple_vat_check(self, country_code, vat_number):
             return bool(self.env['res.country'].search([('code', '=ilike', country_code)]))
         return check_func(vat_number)
 
+    @api.depends('vat', 'country_id')
+    def _compute_vies_vat_to_check(self):
+        """ Retrieve the the VAT number, if one such exists, to be used when checking against the VIES system """
+        eu_country_codes = self.env.ref('base.europe').country_ids.mapped('code')
+        for partner in self:
+            if not partner.vat:
+                partner.vies_vat_to_check = ''
+                continue
+            country_code, vat_split = partner._split_vat(partner.vat)
+            if not country_code.isalpha() and partner.country_id:
+                country_code = partner.country_id.code
+                vat_split = partner.vat
+            partner.vies_vat_to_check = (
+                country_code.upper() in eu_country_codes or
+                country_code.lower() in _region_specific_vat_codes
+            ) and country_code + vat_split or ''
+
+    @api.depends_context('company')
+    @api.depends('vies_vat_to_check')
+    def _compute_perform_vies_check(self):
+        """ Determine whether a VIES check is to be performed on the current VAT number """
+        if self.env.context.get('company_id'):
+            company = self.env['res.company'].browse(self.env.context['company_id'])
+        else:
+            company = self.env.company
+        for partner in self:
+            to_check = partner.vies_vat_to_check
+            if not to_check or to_check[:2].upper() == company.account_fiscal_country_id.code:
+                partner.perform_vies_check = False
+                continue
+            partner.perform_vies_check = company.vat_check_vies
+
     @api.model
     @tools.ormcache('vat')
```

Comment:
Do we still need the `ormcache` here?
We are not likely to re-read the same value again since the result of the check is persisted in the database